src/fetch.rs

```diff
@@ -235,8 +234,8 @@ fn fuzzy_query_registry_index(
     crate_name: impl Into<String>,
     registry: &Url,
 ) -> Result<Vec<CrateVersion>> {
-    let index = crates_index::BareIndex::from_url(registry.as_str())?;
-    let index = index.open_or_clone()?;
+    let mut index = crates_index::Index::from_url(registry.as_str())?;
+    index.update()?;
```

Comment:
wait, why do we update the index here?

Reply:
ah - it's a bug

Reply:
removed in https://github.com/killercup/cargo-edit/pull/533